code/game/machinery/adv_med.dm

```diff
@@ -314,29 +313,37 @@
 	..()
 
 	// shouldn't be reachable if occupant is invalid
-	if (href_list["print"])
-		var/obj/item/paper/R = new(src.loc)
+	if(href_list["print"])
+		var/obj/item/paper/R = new(loc)
 		R.color = "#eeffe8"
-		R.set_content_unsafe("Scan ([src.connected.occupant])", format_occupant_data(src.connected.get_occupant_data()))
+		R.set_content_unsafe("Scan ([connected.occupant])", format_occupant_data(connected.get_occupant_data()))
 
 		print(R, "[src] beeps, printing [R.name] after a moment.")
 
-/obj/machinery/body_scanconsole/ui_interact(mob/user, ui_key = "main", var/datum/nanoui/ui = null, var/force_open = 1, var/datum/topic_state/state = default_state)
-	var/list/data = list()
-	var/occupied = (src.connected && src.connected.occupant)
+/obj/machinery/body_scanconsole/ui_interact(mob/user)
+	var/datum/vueui/ui = SSvueui.get_open_ui(user, src)
+	if (!ui)
+		ui = new(user, src, "medical-bodyscanner", 800, 600, capitalize(name))
+	ui.open()
+
+/obj/machinery/body_scanconsole/vueui_data_change(var/list/data, var/mob/user, var/datum/vueui/ui)
+	if(!data)
+		data = list()
+
 	var/mob/living/carbon/human/occupant
-	if (src.connected)
-		occupant = src.connected.occupant
-
-	data["noscan"]		= src.connected.check_species()
-	data["nocons"]		= !src.connected
-	data["occupied"] 	= occupied
-	data["invalid"]		= src.connected && src.connected.check_species()
-	data["ipc"]			= src.connected && occupant && isipc(occupant)
+	if (connected)
+		occupant = connected.occupant
+
+		VUEUI_SET_CHECK(data["noscan"], !!connected.check_species(), ., data)
```

Comment:
Conditional field setting is a no no. All fields must be present at all times. UI will faill to update if extra root data fields are added afterwards.